internal/credential/registry.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package credential
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/hashicorp/boundary/globals"
+)
+
+// NewFunc is a function that creates a Store from the provided result.
+type NewFunc func(context.Context, *StoreListQueryResult) (Store, error)
```

Comment:
Should this be `NewStoreFunc`? And does this need to be exported? It looks like it is only used by unexported functions/methods.